django/db/models/query.py

```diff
@@ -45,6 +47,33 @@ def __init__(
         self.chunked_fetch = chunked_fetch
         self.chunk_size = chunk_size
 
+    async def _async_generator(self):
+        # Generators don't actually start running until the first time you call
+        # next() on them, so make the generator object in the async thread and
+        # then repeatedly dispatch to it in a sync thread.
+        sync_generator = self.__iter__()
+
+        def next_slice(gen):
+            return list(islice(gen, self.chunk_size))
+
+        while True:
+            chunk = await sync_to_async(next_slice)(sync_generator)
+            for item in chunk:
+                yield item
+            if len(chunk) < self.chunk_size:
+                break
+
+    # __aiter__() is a *synchronous* method that has to then return an
+    # *asynchronous* iterator/generator. Thus, nest an async generator inside
+    # it.
+    # This is a generic iterable converter for now, and is going to suffer a
+    # performance penalty on large sets of items due to the cost of crossing
+    # over the sync barrier each time. Custom __aiter__() methods should be
```

Comment:
Maybe alter this comment to mention _for each chunk_ instead of _each time_? Not sure there's that much value in keeping this whole comment around tbh, seems more like a code review note than something worth keeping in the code base.